ddl/table.go

```diff
@@ -394,3 +394,67 @@ func updateVersionAndTableInfo(t *meta.Meta, job *model.Job, tblInfo *model.Tabl
 	}
 	return ver, t.UpdateTable(job.SchemaID, tblInfo)
 }
+
+func checkPartitionRangeList(meta *model.TableInfo, partName string) error {
+	if meta.Partition != nil {
+		noExist := false
+		for _, def := range meta.Partition.Definitions {
+			if def.Name == partName {
+				noExist = true
+			}
+		}
+		if !noExist {
+			return errors.Trace(infoschema.ErrDropPartitionNonExistent.GenByArgs(partName))
+		}
+	} else {
+		return errors.Trace(infoschema.ErrPartitionMgmtOnNonpartitioned)
+	}
+	return nil
+}
+
+func dropTablePartitionInfo(meta *model.TableInfo, partName string) error {
```

Comment:
What will happen if we drop the last partition?  Will the table degenerate to a normal table?
What happen when we add a partition back? Will the partition id be the same?